app/feature_flags.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/config"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+// setupFeatureFlags called on startup and when the cluster leader changes.
+// Starts or stops the synchronization of feature flags from upstream management.
+func (s *Server) setupFeatureFlags() {
+	s.featureFlagSynchronizerMutex.Lock()
+	defer s.featureFlagSynchronizerMutex.Unlock()
+	license := s.License()
+	inCloud := license != nil && *license.Features.Cloud
+	splitKey := *s.Config().ServiceSettings.SplitKey
+	syncFeatureFlags := inCloud && splitKey != "" && s.IsLeader()
+
+	s.configStore.PersistFeatures(inCloud)
+
+	if syncFeatureFlags {
+		if err := s.startFeatureFlagUpdateJob(); err != nil {
+			s.Log.Error("Unable to setup synchronization with feature flag management. Will fallback to cloud cache.", mlog.Err(err))
```

Comment:
Wondering if this should be a warning. We have an OKR to really remove all non-actionable errors. In this case it will fall back to cloud cache, so there doesn't seem to be anything actionable for an SRE to do.